File: Mine_DS/binarysearch.cc
#include <iostream>
using namespace std;
int BinarySearch(char * arr, int b, int e, char v) {
    if(!arr || b > e)
        return -1;
    int low = b, high = e, mid = 0;
    while(low <= high) {
        mid = low + (high-low)/2;
        if(arr[mid] <= v)
            low = mid + 1;
        else if(arr[mid] > v)
            high = mid - 1;
    }
    if(arr[high] == v)
        return high;
    return -1;
}

int main() {
    char s[] = "23444567";
    cout << BinarySearch(s, 0, 6, '4') << endl;
}

File: Mine_DS/binarysearch.py
import sys

def binary_search(l, b, e, c):
    if not l or b > e:
        return -1
    low, high = b, e
    while low <= high:
        mid = low + (high - low)/2
        if l[mid] < c:
            low = mid + 1
        else:
            high = mid - 1
    if l[high] == c:
        return high
    return -1

if __name__ == "__main__":
    s = "23444567"
    print binary_search(s, 0, len(s)-1, '4')
